geos_ats_package/geos_ats/helpers/restart_check.py

```diff
@@ -79,17 +110,20 @@ def __init__( self,
         assert ( self.atol >= 0.0 )
 
     def filesDiffer( self ):
-        try:
-            with h5py.File( self.file_path, "r" ) as file, h5py.File( self.baseline_path, "r" ) as base_file:
-                self.file_path = file.filename
-                self.baseline_path = base_file.filename
-                self.output.write( "\nRank %s is comparing %s with %s \n" %
-                                   ( MPI.COMM_WORLD.Get_rank(), self.file_path, self.baseline_path ) )
-                self.compareGroups( file, base_file )
-
-        except IOError as e:
-            self.logger.debug( e )
-            self.output.write( str( e ) )
+        # Check to see if the file is on the disk, and wait in case there is any lag in IO
+        file = load_hdf5( self.file_path )
+        base_file = load_hdf5( self.baseline_path )
+        rank = MPI.COMM_WORLD.Get_rank()
+        self.output.write( f"\nRank {rank} is comparing {self.file_path} with {self.baseline_path} \n" )
+
+        # Compare the files
+        if ( file is not None ) and ( base_file is not None ):
+            self.file_path = file.filename
+            self.baseline_path = base_file.filename
+            self.compareGroups( file, base_file )
+
+        else:
+            self.output.write( f"\nRank {rank} Failed to load target and/or baseline files \n" )
```

Comment:
maybe it is useful to know which of the two failed. 

```suggestion
            if ( file is None ):
                self.output.write( f"\nRank {rank} Failed to load target files \n" )
            if ( base_file is None ):
                 self.output.write( f"\nRank {rank} Failed to load baseline files \n" )
```

Reply:
And I wonder if we should add the path info.